File: typescipt/src/compiler/JavaScript/builtInFunctions/endStartDateFunctionCall.ts
import {EndStartDateFunction} from "../../../language/expressions/functions/endStartDateFunction";
import {MethodFunctionCall} from "./methodFunctionCall";
import {CodeWriter} from "../writers/codeWriter";
import {renderExpression} from "../writers/renderExpression";

export abstract class EndStartDateFunctionCall extends MethodFunctionCall {
  public functionNode: EndStartDateFunction;

  protected constructor(functionNode: EndStartDateFunction) {
    super(functionNode);
    this.functionNode = functionNode;
  }

  protected override renderArguments(codeWriter: CodeWriter) {
    renderExpression(this.functionNode.endDateExpression, codeWriter);
    codeWriter.write(", ")
    renderExpression(this.functionNode.startDateExpression, codeWriter);
  }
}

File: src/Lexy.Compiler/Compiler/CSharp/BuiltInFunctions/EndStartDateFunctionCall.cs
using Lexy.Compiler.Compiler.CSharp.Syntax;
using Lexy.Compiler.Language.Expressions.Functions;
using Microsoft.CodeAnalysis;
using Microsoft.CodeAnalysis.CSharp;
using Microsoft.CodeAnalysis.CSharp.Syntax;

namespace Lexy.Compiler.Compiler.CSharp.BuiltInFunctions;

internal abstract class EndStartDateFunctionCall<TExpressionFunction> : MethodFunctionCall<TExpressionFunction>
    where TExpressionFunction : EndStartDateFunction
{
    protected override SeparatedSyntaxList<ArgumentSyntax> GetArguments(TExpressionFunction powerFunction)
    {
        return SyntaxFactory.SeparatedList<ArgumentSyntax>(
            new SyntaxNodeOrToken[]
            {
                SyntaxFactory.Argument(Expressions.ExpressionSyntax(powerFunction.EndDateExpression)),
                SyntaxFactory.Token(SyntaxKind.CommaToken),
                SyntaxFactory.Argument(Expressions.ExpressionSyntax(powerFunction.StartDateExpression))
            });
    }
}